ingester2/src/server/grpc/query.rs

```diff
@@ -335,6 +347,72 @@ fn encode_response(
     })
 }
 
+#[derive(Debug)]
+enum StreamRecorderState {
+    Done,
+    Polling(Option<Box<Span>>),
+}
+
+struct StreamRecorder {
```

Comment:
There's SO MANY STREAMS in the ingester query execution alone! How about a name that says what it records:

```suggestion
struct FlightEncodingLatencyRecorder {
```

(and associated types)